unity-client/Assets/Scripts/MainScripts/DCL/Models/Protocol/ProtocolV2.cs

```diff
@@ -28,34 +28,24 @@ public QuaternionRepresentantion(Quaternion quaternion)
     #region Components
 
     [System.Serializable]
-    public abstract class GenericComponent
+    public class GenericComponent
     {
-
+        public int componentId;
+        public string sharedId;
```

Comment:
what `sharedId` stands for? I think you'd want to name this one `classId`.

Reply:
You are using the name of `classId `in the `BaseDisposable` but I think that it is confusing because you have an enum called CLASS_ID that's refer to the identification of the component itself, however classId into the BaseDisposable refers to a identifier for the sharedId used by the component itself